src/compute-engine/global-types.ts

```diff
@@ -541,15 +541,18 @@ export interface BoxedExpression {
   readonly isPure: boolean;
 
   /**
-   * True if the the value of the expression does not depend on the value of
-   * any other expression.
+   * `True` if this expression's value remains constant.
+   *
+   * If *true* and a function, implies that it is *pure*, and also that all of its arguments are
+   * constant.
    *
-   * For example, a number literal, a symbol with a constant value.
+   * Number literals, symbols with constant values, and numeric functions with constant
+   * subexpressions may all be considered *constant*, i.e.:
    * - `2` is constant
    * - `Pi` is constant
    * - `["Add", "Pi", 2]` is constant
-   * - `x` is not constant
-   * - `["Add", "x", 2]` is not constant
+   * - `x` is inconstant: unless declared with a constant value.
```

Comment:
Perhaps "with a `constant` flag" would be more clear.

Reply:
OK 👍